File: 02linear_list/01array/212_Remove_Duplicates_from_Sorted_Array_II/main.cpp
/*************************************************************************
	> File Name: 211.cpp
	> Author:zhaoyihong 
	> Mail:zhaoyihong_at_126_dot_com 
    > Created Time: 2015年03月16日 星期一 13时36分05秒
 ************************************************************************/

#include  <iostream>
using namespace std;


class Solution {
    public:
        static int removeDuplicates(int A[], int n) 
        {
            if(n <= 2) //允许有2个重复的
            {
                return n;
            }
            
            int index = 1;
            for(int i=2;i<n;++i)
            {
                if(A[index-1] != A[i])
                {
                    A[++index] = A[i];
                }
            }

            return index+1;
        }                    
};

int main(void)
{
    //int a[] = {1,2,2,2,2,2,3,3,4,4,5,5,5,7,8};
    int a[] = {1,2,3};
    int len = Solution::removeDuplicates(a,sizeof(a)/sizeof(*a));
    for(int i=0;i<len;++i)
    {
        cout << a[i] << " ";
    }
    cout << endl;
}

File: 02linear_list/01array/212_Remove_Duplicates_from_Sorted_Array_II/main.py
#!/usr/bin/env python
# coding=utf-8

#注意 python中没有++操作符 , a+=1 也不表示一个数

class Solution:
    # @param a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        n = len(A)
        if n <= 2:
            return n
  
        i = 2
        index = 1
        while i < n:
            if A[index-1] != A[i]:
                index += 1
                A[index] = A[i]
            i += 1

        return index+1

if __name__ == "__main__":

    s = Solution()
#    A = [1,2,2,2,2,3,4,5,5,7]
    A=[1,2,3]
    len = s.removeDuplicates(A)
    print len
    print A[:len]


